src/EditorFeatures/CSharpTest/MakeClassAbstract/MakeClassAbstractTests.cs

```diff
@@ -0,0 +1,273 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis.CodeFixes;
+using Microsoft.CodeAnalysis.CSharp.MakeClassAbstract;
+using Microsoft.CodeAnalysis.Diagnostics;
+using Microsoft.CodeAnalysis.Editor.CSharp.UnitTests.Diagnostics;
+using Microsoft.CodeAnalysis.Test.Utilities;
+using Xunit;
+
+namespace Microsoft.CodeAnalysis.Editor.CSharp.UnitTests.MakeClassAbstract
+{
+    public class MakeClassAbstractTests : AbstractCSharpDiagnosticProviderBasedUserDiagnosticTest
+    {
+        internal override (DiagnosticAnalyzer, CodeFixProvider) CreateDiagnosticProviderAndFixer(Workspace workspace)
+            => (null, new CSharpMakeClassAbstractCodeFixProvider());
+
+        [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsMakeClassAbstract)]
+        public async Task TestMethod()
+        {
+            await TestInRegularAndScript1Async(
```

Comment:
Can you add test for type with partial declarations? Should we add abstract modifier to all declarations or just the containing declaration from trigger diagnostic location?

Reply:
It's failing for `partial` but for me the issue comes from the `WithModifiers` method not doing a proper merge. The `SyntaxGenerator.Merge` method returns `public abstract` when merging `public partial` and `public abstract`. IMO it should return `public partial abstract`.

Reply:
Would it be possible to create a separate PR to fix the SyntaxGenerator, and wait for that to go in before taking this up? Alternatively, can you file an issue for the SyntaxGenerator merge, add the test with partial declarations and mark it as skipped with that issue? I am fine with either approach that works for you.

Reply:
I would prefer the skip that would avoid too many extra master merges. I will create the ticket. Are you ok with keeping the `WithModifiers` or what is your idea?

Reply:
I think it is fine.

Reply:
#41654